src/Sarif/Visitors/RunMergingVisitor.cs

```diff
@@ -78,7 +78,10 @@ public override Result VisitResult(Result node)
             }
 
             // Cache RuleId and set Result.RuleIndex to the (new) index
-            string ruleId = node.ResolvedRuleId(CurrentRun);
+            string resultRuleId = node.ResolvedRuleId(CurrentRun);
+            ReportingDescriptor rule = node.GetRule(CurrentRun);
+            string ruleId = rule?.Id ?? resultRuleId;
```

Comment:
>string ruleId = rule?.Id ?? resultRuleId; [](http://example.com/codeflow?start=12&length=41)

Is there a case where rule.Id might not be null and resultRuleId may be another value?